### PR TITLE
feat(feature-store): Implement Feature Store management page

### PR DESCRIPTION
**Pull request by ntkathole**
https://redhat.atlassian.net/browse/RHOAIENG-61263 
https://redhat.atlassian.net/browse/RHOAIENG-76741 

## Description

Implements the Feature Store management list page (`/develop-train/feature-store/manage`) with the following capabilities:
- **Sortable table** — Columns for Name, Namespace, Status, Version, and Created, with sort support on Name, Namespace, Status, and Created.
- **Expandable detail rows** — Each row expands to show a summary of the feature store's Feast project, registry type, online/offline store configuration, authorization, scaling, CronJob schedule, batch engine, client ConfigMap, service hostnames, and conditions.
- **Delete action** — Kebab menu with SSAR-gated delete action that opens a confirmation modal. 404 responses during deletion are treated as success (CR already removed).
- **Empty state** — When no feature stores exist, shows an empty state with a "Create feature store" button (if the user has `create` permission).
- **RBAC-protected route** — The `/manage` route is gated by `list` permission on the `FeatureStore` model via `accessAllowedRouteHoC`.
- **Updated CoreLoader empty state** — Replaced the external OpenShift Console link with in-app "Create feature store" and "Manage feature stores" buttons. Updated user-facing copy for non-admin users.

Also, Added a `featureStoreAdmin` feature flag (defaulting to false / disabled) to gate the Feature Store admin UI 

## How Has This Been Tested?

- 27 unit tests covering all table row states, expanded detail summaries (registry, online/offline stores, authz, scaling, cronJob, batch engine, client ConfigMap), status phase labels, Primary badge, and service hostnames.
- Built and deployed to OpenShift cluster (`quay.io/nkathole/odh-dashboard:[RHOAIENG-61263](https://redhat.atlassian.net/browse/RHOAIENG-61263)`) and manually verified the manage page, table rendering, row expansion, and delete flow.
- TypeScript type-check and ESLint pass clean.

## Test Impact
- Added `FeatureStoreTableRow.spec.tsx` with 27 tests covering all summary helper functions (`phaseLabel`, `getRegistrySummary`, `getOnlineStoreSummary`, `getOfflineStoreSummary`, `getAuthzSummary`, `getScalingSummary`), expanded/collapsed states, status labels, Primary badge, service hostnames, conditions, cronJob, batch engine, and client ConfigMap rendering.

@ralombar Added Feature Store Manage page

## Screenshots

Screenshot 2026-07-13 at 3 28 46 PM

Screenshot 2026-07-13 at 3 28 02 PM

Screenshot 2026-07-13 at 3 59 00 PM

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

## Summary by CodeRabbit

* **New Features**
  * Added a **Feature Store Admin** manage experience with a sortable list, expandable row details (including version), and delete confirmation.
  * Introduced a feature-flagged **Feature Store Admin** UI area for the create/manage flow, protected by area availability and access checks.
* **Bug Fixes**
  * Updated the “no feature stores” empty state messaging/CTAs for admins vs non-admins, including an optional external link for some admin configurations.
  * Improved delete handling so “not found” deletions are treated as successful.
* **Tests**
  * Added/extended Jest tests covering the manage page, table row rendering/expansion, and delete modal edge cases.

[RHOAIENG-61263]: https://redhat.atlassian.net/browse/RHOAIENG-61263?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dpanshug** [comment]
@odh-dashboard-agent preflight check

**Comment by ralombar** [comment]
**Screenshot 1 -** 
- Heading should be (unless its already had a content review): No feature stores yet 
- Body: To get started, create a feature store.
- With the manage feature stores button - remove the secondary button if there are no feature stores as it will be empty. 
I also noticed the text under Feature store overview is outdated, it should be:
Feature store repositories connect models to data by enabling you to store, manage, and serve ML features from your existing data sources. To consume and manage resources within feature store repositories , connect repos with your workbenches. Learn how to connect. 

See first row in the table https://docs.google.com/document/d/1dqOk7Zc984pBTxaEmhcprX9vonXNHsbFjk434in39a8/edit?usp=sharing 
However, let me know if what is in the screenshot is correct for this scenario, and has already been reviewed. 

**Screenshot 3 -** 
Date format in the created column
- Should look like Dec 6, 2026, 9:07 AM
- Uppercase AM in the date format.

File: packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    data-testid={props['data-testid']}
+  >
+    Create feature store
+  </Button>
+);
+
+const FeatureStoreListPage: React.FC = () => {
+  const { featureStores, loaded, error, refresh } = useExistingFeatureStores();
+  const [canCreate] = useAccessAllowed(verbModelAccess('create', FeatureStoreModel));
+  const [canDelete] = useAccessAllowed(verbModelAccess('delete', FeatureStoreModel));
```
[the remark this answers is not in the document]
Comment (ntkathole) [reply packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx:89]: The hook accepts a single AccessReviewResourceAttributes and returns a [boolean, boolean] tuple — it doesn't support batching multiple verbs. The underlying AccessReviewContext does handle caching and deduplication internally (via accessReviewCache), so the two calls already share the same cache and context provider; they just trigger two separate canIAccess lookups.

Changing this would require modifying the shared useAccessAllowed hook signature (or creating a new multi-verb variant), which is outside the scope of this feature work and would affect all consumers across the dashboard. Given it's only 2 calls and the cache layer already prevents redundant network requests for the same resource attributes, the practical impact is minimal. I'd leave this as-is.

File: packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      >
+        <Table
+          data-testid="feature-store-list-table"
+          id="feature-store-list-table"
+          enablePagination
+          data={featureStores}
+          columns={columns}
+          defaultSortColumn={1}
+          toolbarContent={toolbarContent}
+          emptyTableView={<DashboardEmptyTableView onClearFilters={() => undefined} />}
```
[the remark this answers is not in the document]
Comment (ntkathole) [reply packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx:157]: The emptyTableView with its no-op onClearFilters is dead code — there's no filter/search control in the toolbar, and the true empty state is already handled by the emptyStatePage prop on ApplicationsPage. However, I'd recommend keeping it as a defensive placeholder rather than removing it. If a search/filter is added to the toolbar later, the DashboardEmptyTableView will be needed, and removing it now means someone would have to rediscover the pattern and add it back. The no-op handler is harmless, and this is consistent with how other list pages in the dashboard are structured.

**Comment by ntkathole** [comment]
> **Screenshot 1 -**
> 
>     * Heading should be (unless its already had a content review): No feature stores yet
> 
>     * Body: To get started, create a feature store.
> 
>     * With the manage feature stores button - remove the secondary button if there are no feature stores as it will be empty.
>       I also noticed the text under Feature store overview is outdated, it should be:
>       Feature store repositories connect models to data by enabling you to store, manage, and serve ML features from your existing data sources. To consume and manage resources within feature store repositories , connect repos with your workbenches. Learn how to connect.
> 
> 
> See first row in the table https://docs.google.com/document/d/1dqOk7Zc984pBTxaEmhcprX9vonXNHsbFjk434in39a8/edit?usp=sharing However, let me know if what is in the screenshot is correct for this scenario, and has already been reviewed.
> 
> **Screenshot 3 -** Date format in the created column
> 
>     * Should look like Dec 6, 2026, 9:07 AM
> 
>     * Uppercase AM in the date format.

@ralombar Updated the code and screenshots, please re-visit

**Comment by ralombar** [comment]
> > **Screenshot 1 -**
> > ```
> > * Heading should be (unless its already had a content review): No feature stores yet
> > 
> > * Body: To get started, create a feature store.
> > 
> > * With the manage feature stores button - remove the secondary button if there are no feature stores as it will be empty.
> >   I also noticed the text under Feature store overview is outdated, it should be:
> >   Feature store repositories connect models to data by enabling you to store, manage, and serve ML features from your existing data sources. To consume and manage resources within feature store repositories , connect repos with your workbenches. Learn how to connect.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > See first row in the table https://docs.google.com/document/d/1dqOk7Zc984pBTxaEmhcprX9vonXNHsbFjk434in39a8/edit?usp=sharing However, let me know if what is in the screenshot is correct for this scenario, and has already been reviewed.
> > **Screenshot 3 -** Date format in the created column
> > ```
> > * Should look like Dec 6, 2026, 9:07 AM
> > 
> > * Uppercase AM in the date format.
> > ```
> 
> @ralombar Updated the code and screenshots, please re-visit

LGTM!! @ntkathole

File: packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  <Button
+    variant="primary"
+    component={(linkProps: React.ComponentProps<'a'>) => (
+      <Link {...linkProps} to="/develop-train/feature-store/create" />
+    )}
+    icon={<PlusCircleIcon />}
+    data-testid={props['data-testid']}
+  >
+    Create feature store
+  </Button>
```
[the remark this answers is not in the document]
Comment (ntkathole) [reply packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx:83]: It's the package that owns all Feature Store UI functionality, other packages follow the same pattern.

File: packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
```
Comment (dpanshug) [inline packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx]: Disable this button for now

File: packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+  },
+];
+
+const CreateFeatureStoreButton: React.FC<{ 'data-testid'?: string }> = (props) => (
+  <Button
+    variant="primary"
+    component={(linkProps: React.ComponentProps<'a'>) => (
+      <Link {...linkProps} to="/develop-train/feature-store/create" />
+    )}
+    icon={<PlusCircleIcon />}
```
Comment (dpanshug) [inline packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx]: We don't usually use + icon for create buttons. cc @ralombar 

Image

File: packages/feature-store/src/FeatureStoreRoutes.tsx [hunk 6]
```diff
@@ -33,8 +39,14 @@ export const featureRoute = (
...
 
+const AreaGatedManagePage = conditionalArea(
+  SupportedArea.FEATURE_STORE_ADMIN,
+  true,
+)(accessAllowedRouteHoC(verbModelAccess('list', FeatureStoreModel))(FeatureStoreListPage));
+
 const FeatureStoreRoutes: React.FC = () => (
   <Routes>
+    <Route path="manage" element={<AreaGatedManagePage />} />
     <Route
```
Comment (dpanshug) [inline packages/feature-store/src/FeatureStoreRoutes.tsx:50]: If you navigate to `/develop-train/feature-store/manage/anything` it shows empty page

```suggestion
const FeatureStoreRoutes: React.FC = () => (
<Routes>
    <Route path="manage/*" element={<AreaGatedManagePage />} />
<Route
```

**Review by dpanshug — CHANGES_REQUESTED** [review]
* AGENTS.md "Read-only UI" constraint not updated

**Comment by dpanshug** [comment]
@ntkathole can you rebase with main

**Comment by ntkathole** [comment]
> @ntkathole can you rebase with main

done

**Review by claudialphonse78 — CHANGES_REQUESTED** [review]
Could you resolve the linting issues

File: packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx (hunk 5 above)
Comment (ntkathole) [reply packages/feature-store/src/screens/manage/FeatureStoreListPage.tsx]: removed, make it similar to other buttons

**Comment by ntkathole** [comment]
seems CI failures not related to this PR

**Review by dpanshug — APPROVED** [review]
/lgtm - CI failures are not related
